### ISSUE TITLE
Don't `ensure_table_uri` when creating a table `with_log_store`

### ISSUE THREAD

**Issue by gruuya** [delta-io/delta-rs#2036]
# Description

Currently when a `CreateBuilder` is provided with a log store, it will try to create a URL from it's root (via `ensure_table_uri`), and then re-convert that to a `String`
https://github.com/delta-io/delta-rs/blob/9264edea89a2fc1c35f4a6b9faab125748ff3651/crates/deltalake-core/src/operations/create.rs#L222-L226

Besides potentially create missing dirs in the case of local FS, I don't think this round trip is actually needed in other cases (e.g. cloud URLs).

The issue this leads to for us is that `ensure_table_uri` in turn calls `resolve_uri_type`, which errors out if the URL scheme is not for one of the registered object store factories.

**Use Case**

I'd like to be able to use `with_log_store` on the `CreateBuilder` without having to register an object store factory beforehand.

In other words I'd like to be able to re-use the object/log store that I keep an `Arc` to without re-creating one from scratch every time.

**Related Issue(s)**

**Comment by gruuya** [comment]
I guess one resolution to this would be to have my `ObjectStoreFactory` just return the pre-configured ones according to the URL, which is also fine.

**Comment by rtyler** [comment]
@gruuya I see you're following `main` closely :laughing: Are you working on a different storage integration, or just concerned with the goofy file creation behavior in core here?

**Comment by gruuya** [comment]
> I see you're following main closely

Yeah, pretty nice to see the pace of development here, kudos :)

> Are you working on a different storage integration, or just concerned with the goofy file creation behavior in core here?

I was mostly confused by how previously I had an object/log store instance that I'd supply to the `CreateBuilder` and be done with it, whereas now I need to register the factory as well.

Come to think of it, this mechanism not only makes sense broadly speaking but it's a pretty good idea actually, given that I've been contemplating how to introduce dynamic object/log store creation down the road.

Sorry for the noise and thanks!
